src/v0/destinations/sfmc/transform.js

```diff
@@ -224,7 +225,16 @@ const responseBuilderSimple = async (message, category, destination) => {
     throw new ConfigurationError('Creating or updating contacts is disabled');
   }
 
-  if (category.type === 'track' && hashMapExternalKey[message.event.toLowerCase()]) {
+  if (category.type === 'track') {
+    if (isEmpty(message.event)) {
+      throw new ConfigurationError('Event name is required for this track events');
+    }
+    if (!isDefinedAndNotNull(hashMapExternalKey[message.event.toLowerCase()])) {
+      throw new ConfigurationError('Event not mapped for this track call');
+    }
+    if (typeof message.event !== 'string') {
```

Comment:
Addressed

Reply:
Let's have these validations (`if (isEmpty(message.event)) {` & `if (typeof message.event !== 'string') {`) in `processEvent` function rather than here, WDYT ?

The other validation(`!isDefinedAndNotNull(hashMapExternalKey[message.event.toLowerCase()])`) let it be here.